File: API/Features/Sales/Transactions/Dtos/Read/SaleReadDto.cs
using System;
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Sales.Transactions {

    public class SaleReadDto : IMetadata {

        public Guid InvoiceId { get; set; }
        public DateTime Date { get; set; }
        public DateTime TripDate { get; set; }
        public int InvoiceNo { get; set; }
        public decimal NetAmount { get; set; }
        public decimal VatPercent { get; set; }
        public decimal VatAmount { get; set; }
        public decimal GrossAmount { get; set; }
        public string Remarks { get; set; }
        public bool IsEmailSent { get; set; }
        public bool IsCancelled { get; set; }
        public SaleFormAadeVM Aade { get; set; }
        public SimpleEntity Customer { get; set; }
        public SaleFormDocumentTypeVM DocumentType { get; set; }
        public SimpleEntity PaymentMethod { get; set; }
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/sales/invoices/classes/dtos/form/sale-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { CustomerAutoCompleteVM } from 'src/app/features/sales/customers/classes/view-models/customer-autocomplete-vm'
import { DocumentTypeVM } from '../../view-models/form/documentType-vm'
import { ItemReadDto } from './item-read-dto'
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface SaleReadDto extends Metadata {

    invoiceId: Guid
    date: string
    invoiceNo: number
    customer: CustomerAutoCompleteVM 
    documentType: DocumentTypeVM
    paymentMethod: SimpleEntity
    items: ItemReadDto[]
    remarks: string
    isEmailSent: boolean
    isCancelled: boolean
    netAmount: number
    vatPercent: number
    vatAmount: number
    grossAmount: number

}
